core/src/test/java/ai/timefold/solver/core/impl/domain/solution/mutation/MutationCounterTest.java

```diff
@@ -116,40 +115,48 @@ void countMutationsAll() {
+        var mutationCounter = new MutationCounter<>(solutionDescriptor);
 
-        TestdataValue val1 = new TestdataValue("1");
-        TestdataValue val2 = new TestdataValue("2");
-        List<TestdataValue> valueList = Arrays.asList(val1, val2);
+        var val1 = new TestdataValue("1");
+        var val2 = new TestdataValue("2");
+        var valueList = Arrays.asList(val1, val2);
 
-        int entityListSize = 3;
-        int subEntityListSize = 7;
-        int rawEntityListSize = 17;
+        var entityListSize = 3;
+        var subEntityListSize = 7;
+        var rawEntityListSize = 17;
 
-        TestdataExtendedEntitySolution a =
-                TestdataExtendedEntitySolution.generateSolution(entityListSize, subEntityListSize, rawEntityListSize);
+        var a = TestdataBothAnnotatedExtendedSolution.generateSolution(entityListSize, subEntityListSize, rawEntityListSize);
         a.setValueList(valueList);
         a.getEntity().setValue(val1);
         a.getSubEntity().setValue(val1);
+        a.getSubEntity().setSubValue(val1);
         a.getEntityList().forEach(e -> e.setValue(val1));
-        a.getSubEntityList().forEach(e -> e.setValue(val1));
+        a.getSubEntityList().forEach(e -> {
+            e.setValue(val1);
+            e.setSubValue(val1);
+        });
         for (Object o : a.getRawEntityList()) {
-            ((TestdataEntity) o).setValue(val1);
+            ((TestdataBothAnnotatedChildEntity) o).setValue(val1);
+            ((TestdataBothAnnotatedChildEntity) o).setSubValue(val1);
         }
 
-        TestdataExtendedEntitySolution b =
-                TestdataExtendedEntitySolution.generateSolution(entityListSize, subEntityListSize, rawEntityListSize);
+        var b = TestdataBothAnnotatedExtendedSolution.generateSolution(entityListSize, subEntityListSize, rawEntityListSize);
         b.setValueList(valueList);
         b.getEntity().setValue(val2);
         b.getSubEntity().setValue(val2);
+        b.getSubEntity().setSubValue(val2);
         b.getEntityList().forEach(e -> e.setValue(val2));
-        b.getSubEntityList().forEach(e -> e.setValue(val2));
+        b.getSubEntityList().forEach(e -> {
+            e.setValue(val2);
+            e.setSubValue(val2);
+        });
         for (Object o : b.getRawEntityList()) {
-            ((TestdataEntity) o).setValue(val2);
+            ((TestdataBothAnnotatedChildEntity) o).setValue(val2);
+            ((TestdataBothAnnotatedChildEntity) o).setSubValue(val2);
         }
-
-        assertThat(mutationCounter.countMutations(a, b)).isEqualTo(entityListSize + subEntityListSize + rawEntityListSize + 2);
+        var baseDescriptorCount = entityListSize + subEntityListSize + rawEntityListSize + 2;
+        var childDescriptorCount = subEntityListSize * 2 + rawEntityListSize * 2 + 2;
+        assertThat(mutationCounter.countMutations(a, b)).isEqualTo(baseDescriptorCount + childDescriptorCount);
```

Comment:
I do not expect tests to change behavior.
If the test needs to change, then the domains were not duplicates.

Reply:
I could simply move the extended classes to the inheritance package, but I attempted to merge some models with similar features. Adding new planning variables to the existing inheritance classes also changed the score, as the constraint was evaluated more times than before.

Reply:
One solution is to restore the extended classes and merge the packages. What do you think?

Reply:
Merging models is fine, I like that we're removing boilerplate code. 
But why does that mean changing behavior?